main.tf

```diff
@@ -244,83 +265,56 @@ resource "azurerm_mssql_server_vulnerability_assessment" "va_secondary" {
-  name                = element(var.firewall_rules, count.index).name
-  resource_group_name = local.resource_group_name
-  server_name         = azurerm_sql_server.primary.name
-  start_ip_address    = element(var.firewall_rules, count.index).start_ip_address
-  end_ip_address      = element(var.firewall_rules, count.index).end_ip_address
+resource "azurerm_mssql_firewall_rule" "fw01" {
+  count            = var.enable_firewall_rules && length(var.firewall_rules) > 0 ? length(var.firewall_rules) : 0
+  name             = element(var.firewall_rules, count.index).name
+  start_ip_address = element(var.firewall_rules, count.index).start_ip_address
+  end_ip_address   = element(var.firewall_rules, count.index).end_ip_address
+  server_id        = azurerm_mssql_server.primary.id
+
 }
 
-resource "azurerm_sql_firewall_rule" "fw02" {
-  count               = var.enable_failover_group && var.enable_firewall_rules && length(var.firewall_rules) > 0 ? length(var.firewall_rules) : 0
-  name                = element(var.firewall_rules, count.index).name
-  resource_group_name = local.resource_group_name
-  server_name         = azurerm_sql_server.secondary.0.name
-  start_ip_address    = element(var.firewall_rules, count.index).start_ip_address
-  end_ip_address      = element(var.firewall_rules, count.index).end_ip_address
+resource "azurerm_mssql_firewall_rule" "fw02" {
+  count            = var.enable_failover_group && var.enable_firewall_rules && length(var.firewall_rules) > 0 ? length(var.firewall_rules) : 0
+  name             = element(var.firewall_rules, count.index).name
+  start_ip_address = element(var.firewall_rules, count.index).start_ip_address
+  end_ip_address   = element(var.firewall_rules, count.index).end_ip_address
+  server_id        = azurerm_mssql_server.secondary[*].id
 }
 
 #---------------------------------------------------------
 # Azure SQL Failover Group - Default is "false"
 #---------------------------------------------------------
 
-resource "azurerm_sql_failover_group" "fog" {
-  count               = var.enable_failover_group ? 1 : 0
-  name                = "sqldb-failover-group"
-  resource_group_name = local.resource_group_name
-  server_name         = azurerm_sql_server.primary.name
-  databases           = [azurerm_sql_database.db.id]
-  tags                = merge({ "Name" = format("%s", "sqldb-failover-group") }, var.tags, )
+resource "azurerm_mssql_failover_group" "fog" {
+  count     = var.enable_failover_group ? 1 : 0
+  name      = "sqldb-failover-group"
+  databases = [azurerm_mssql_database.db.id]
+  tags      = merge({ "Name" = format("%s", "sqldb-failover-group") }, var.tags, )
+  server_id = azurerm_mssql_server.primary.id
 
-  partner_servers {
-    id = azurerm_sql_server.secondary.0.id
+  partner_server {
+    id = azurerm_mssql_server.secondary[0].id
   }
 
   read_write_endpoint_failover_policy {
     mode          = "Automatic"
     grace_minutes = 60
   }
-
-  readonly_endpoint_failover_policy {
```

Comment:
Is there no alternative for this?

Reply:
this is deprecated
used this readonly_endpoint_failover_policy_enabled